### ISSUE TITLE
Utility types from Effector

### ISSUE THREAD

**Issue by dimensi** [zerobias/effector#369]
## Proposal

There is a lack of type-helper exports in the effector.

## Use case

For example, I could really use a helper that would allow me to output the Store content type:

```ts
type GetStoreValue<T> = T extends Store<infer S> ? S : never;

const $store = createStore({ limit: 0, offset: 0 })
const fxGetPage = createEffect<GetStoreValue<typeof $store>, IResult, IError>({
   handler(params) {
     params // { limit: number, offset: number }
   }
})
```

**Comment by zerobias** [comment]
Maybe we also can provide helpers for effects? https://github.com/zerobias/effector/blob/master/packages/effector/index.d.ts#L710

**Comment by dimensi** [comment]
To show what can come from effect the store?
```
const fxEffect = createEffect<Params, { data: Response }>()

const store = createStore<GetDone<typeof fxEffect>>()
```

**Comment by zerobias** [comment]
yes, at least I remember some requests for that helper too

**Comment by zerobias** [comment]
Implemented in [effector@21.1.0](https://github.com/zerobias/effector/releases/tag/effector%4021.1.0)
